### PR TITLE
Setup lint config and .gitattributes

### PR DESCRIPTION
**Pull request by antoncoding**

### CODE REVIEW

**Comment by antoncoding** [comment]
Rules for `ts` and `js` files: 
`.eslintrc.json` && `prettierrc.json`

Rules for `.sol` files
`.solhint.json` && override section in `prettierrc.json`

File: package.json [hunk 1]
```diff
@@ -44,11 +44,7 @@
     "compile:types": "typechain --target=truffle --outDir build/types/truffle-types \"build/contracts/*.json\"",
     "deploy:development": "npm run build && npm run migrate -- --network development",
     "deploy:kovan": "npm run build && npm run migrate -- --network kovan",
-    "compile:contracts": "truffle compile",
-    "compile:types": "typechain --target=truffle --outDir build/types/truffle-types \"build/contracts/*.json\"",
-    "build": "npm run compile:contracts && npm run compile:types",
     "pretest": "npm run build",
```
Comment (haythem96) [inline package.json:47]: Should we remove this `pretest` command ? it just run the `build`

File: package.json [hunk 2]
```diff
@@ -44,11 +44,7 @@
...
-    "test": "truffle test",
     "test:kovan": "truffle test --network kovan",
```
Comment (haythem96) [inline package.json:48]: I think also we will not need `test:kovan`, unit tests will be executed locally, right ?

File: package.json (hunk 1 above)
Comment (antoncoding) [reply package.json:47]: Okay to me

File: .prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "overrides": [
+    {
+      "files": "contracts/**/*.sol",
+      "options": {
+        "tabWidth": 4,
+        "singleQuote": false
+      }
+    }
+  ]
+}
```
Comment (aparnakr) [inline .prettierrc.json:16]: does this auto adjust the code or is this more a check?

Comment (antoncoding) [reply .prettierrc.json:16]: The default part (for ts) is included in the eslint as an extension, so we will run auto lint in `lint-stage` with `eslint --fix`

The overrides part for .sol files will also be run in `lint-stage` with `prettier --write`

```json
# package.json
"lint-staged": {
    "**/*.sol": [
      "prettier --write",
      "git add"
    ],
    "test/**": [
      "eslint --fix",
      "git add"
    ]
  },
```

**Pull request merged**